### ISSUE TITLE
oas3-valid-schema-example Rule passes with invalid example object

### ISSUE THREAD

**Issue by yashwason** [daveshanley/vacuum#392]
I have the oas3-valid-schema-example enabled on my Vacuum configuration. It rightly warns about a missing example on the below 'address' field if I delete `examples`. But if I modify the example e.g. I delete the required fields `country` and `address1`; the lint still passes.

```
ResponseAddressFields:
    type: object
    description: Address-related fields of response body
    properties:
        address:
            type: object
            description: Primary physical address of the organisation
            required:
                - address1
                - city
                - country
            examples: 
                - address1: J-234 Gandhi Nagar
                  address2: Street number 2, Block 2
                  city: Delhi
                  subDivision:
                    entity: countrySubDivision
                    name: Delhi
                    type: union territory
                    code: IN-DL
                  country:
                    entity: country
                    name: India
                    code: IN
                  postalCode: '110019'
            allOf:
                - $ref: '#/components/schemas/CommonAddressFields'
                - type: object
                  properties:
                      subDivision:
                          type: object
                          required: 
                            - entity
                            - name
                            - type
                            - code
                          properties:
                              entity:
                                  type: string
                              name:
                                  type: string
                              type:
                                  type: string
                              code:
                                  $ref: '#/components/schemas/SubDivisionCodeField'
    
                      country:
                          type: object
                          required: 
                            - entity
                            - name
                            - code
                          properties:
                              entity:
                                  type: string
                              name:
                                  type: string
                              code:
                                  $ref: '#/components/schemas/CountryCodeField'
```

**Comment by daveshanley** [comment]
This should be resolved in `v0.7.0` can you try again?

**Comment by yashwason** [comment]
Resolved indeed. Vacuum now warns about missing properties in examples.
